ocular/src/chain/client/automated_tx_handler.rs

```diff
@@ -0,0 +1,292 @@
+                return Err(AutomatedTxHandlerError::TomlError(err.to_string()));
+            }
+        };
+
+        dbg!(&toml);
+
+        // Do some key processing/manipulation
+        let pem =
+            match pkcs8::PrivateKeyDocument::read_pem_file(toml.sender.source_private_key_path) {
+                Ok(res) => res,
+                Err(err) => return Err(AutomatedTxHandlerError::KeyHandlingError(err.to_string())),
+            };
+
+        let signing_key: k256::elliptic_curve::SecretKey<k256::Secp256k1> = pem
+            .to_pem(LineEnding::default())
+            .expect("Could not convert to pem.")
+            .parse()
+            .expect("Could not parse pem");
+
+        let signing_key_bytes = signing_key.to_bytes();
+
+        let verifying_key = SigningKey::from_bytes(&signing_key_bytes)
+            .expect("Could not create verifying key from signing key.")
+            .public_key();
+
+        let account_id = verifying_key
+            .account_id(toml.sender.prefix)
+            .expect("Could not get account id from verifying key.");
+
+        let mut response_vec: Vec<Response> = Vec::new();
+
+        for tx in toml.transactions.iter() {
+            let recipient_account_id = match AccountId::from_str(tx.destination_account) {
+                Ok(res) => res,
+                Err(err) => return Err(AutomatedTxHandlerError::KeyHandlingError(err.to_string())),
+            };
+
+            let response = match self
+                .send(
+                    Account {
+                        id: account_id.clone(),
+                        public_key: verifying_key,
+                        private_key: SigningKey::from_bytes(&signing_key_bytes)
+                            .expect("Could not create signing key."),
+                    },
+                    recipient_account_id,
+                    Coin {
+                        denom: tx.denom.parse().expect("Could not parse denom."),
+                        amount: tx.amount.into(),
+                    },
+                    TxMetadata {
+                        chain_id: toml
+                            .sender
+                            .chain_id
+                            .parse()
+                            .expect("Could not parse chain id"),
+                        account_number: tx.account_number,
+                        sequence_number: tx.sequence_number,
+                        gas_limit: tx.gas_limit,
+                        timeout_height: tx.timeout_height,
+                        memo: tx.memo.to_string(),
+                    },
+                )
+                .await
+            {
+                Ok(res) => res,
+                Err(err) => return Err(AutomatedTxHandlerError::TxError(err.to_string())),
+            };
+
+            // TODO: What if we terminate early, what happens to previous results?
+            // Do we want to save this differently
+            response_vec.push(response);
```

Comment:
perhaps a tx history dump in .ocular?

Reply:
Yup, that sounds like a good idea!

Reply:
Added in https://github.com/PeggyJV/ocular/issues/21